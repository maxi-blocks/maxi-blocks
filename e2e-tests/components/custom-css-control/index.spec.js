/**
 * WordPress dependencies
 */
import { createNewPost, pressKeyWithModifier } from '@wordpress/e2e-test-utils';

/**
 * Internal dependencies
 */
import {
	getBlockStyle,
	openSidebarTab,
	addCustomCSS,
	changeResponsive,
	getAttributes,
	insertMaxiBlock,
	updateAllBlockUniqueIds,
} from '../../utils';

describe('Custom-Css-Control', () => {
	it('Checking the custom-css', async () => {
		await createNewPost();
		await insertMaxiBlock(page, 'Group Maxi');
		await updateAllBlockUniqueIds(page);
		await expect(await addCustomCSS(page)).toMatchSnapshot();
	}, 500000);

	it('Checking the custom-css responsive', async () => {
		await changeResponsive(page, 's');

		// check base value in S responsive
		const baseValue = await page.$$eval(
			'.w-tc-editor textarea',
			input => input[0].value
		);

		expect(baseValue).toStrictEqual('background: red');

		// change S value
		await page.$eval('.w-tc-editor textarea', input => input.focus());

		await pressKeyWithModifier('primary', 'a');
		await page.keyboard.type('background: blue');
		await page.waitForTimeout(200);

		// change xs responsive
		await changeResponsive(page, 'xs');

		const xsValue = await page.$$eval(
			'.w-tc-editor textarea',
			input => input[0].value
		);

		expect(xsValue).toStrictEqual('background: blue');

		// change m responsive
		await changeResponsive(page, 'm');

		const mValue = await page.$$eval(
			'.w-tc-editor textarea',
			input => input[0].value
		);

		expect(mValue).toStrictEqual('background: red');
	});

	it('Checking the custom-css Validation', async () => {
		const accordionPanel = await openSidebarTab(
			page,
			'advanced',
			'custom css'
		);

		const selector = await accordionPanel.$(
			'.maxi-custom-css-control__category select'
		);

		await selector.select('group');

		await page.$eval('.maxi-custom-css-control__group textarea', input =>
			input.focus()
		);

		await page.keyboard.type('VALIDATE ERROR');
		await page.waitForTimeout(200);
		await page.waitForSelector('.maxi-custom-css-control__group button');

		await accordionPanel.$eval(
			'.maxi-custom-css-control__group button',
			button => button.click()
		);
		await page.waitForTimeout(150);

		const error = await accordionPanel.$$eval(
			'.maxi-custom-css-control__error',
			input => input[0].innerText
		);
		expect(error).not.toBe('Valid');

		// focus out
		await page.$eval('.editor-post-title__input', input => input.focus());

		// await validation
		await page.waitForTimeout(1000);
		await page.$eval('.maxi-group-block', block => block.focus());

		// return css block
		expect(await getAttributes('custom-css-general')).toMatchSnapshot();
		expect(await getAttributes('custom-css-m')).toMatchSnapshot();
		expect(await getAttributes('custom-css-s')).toMatchSnapshot();

		expect(await getBlockStyle(page)).toMatchSnapshot();
	});
});
