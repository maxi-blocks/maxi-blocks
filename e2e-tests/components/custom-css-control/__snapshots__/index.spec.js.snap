// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom-Css-Control Checking the custom-css 1`] = `"<style>body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block{ background: red;}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block::before,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block::before{ background: red;}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block::after,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block::after{ background: red;}</style>"`;

exports[`Custom-Css-Control Checking the custom-css Validation 1`] = `"<style>body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block{ background: red;}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block::before,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block::before{ background: red;}body.maxi-blocks--active .edit-post-visual-editor .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block::after,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive] .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block::after{ background: red;}body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\\"m\\"] .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\\"s\\"] .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\\"xs\\"] .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block{ background: red;}body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\\"s\\"] .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\\"xs\\"] .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block{ background: red;}body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\\"s\\"] .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block::after,body.maxi-blocks--active .edit-post-visual-editor[maxi-blocks-responsive=\\"xs\\"] .maxi-block.maxi-block--backend.group-maxi-12.maxi-group-block::after{ background: blue;}</style>"`;
